test/unit/evault/modules/Vault/batch.t.sol

```diff
@@ -0,0 +1,345 @@
+        oracle.setPrice(address(eTST2), unitOfAccount, 0.4e18);
+
+        IEVC.BatchItem[] memory items = new IEVC.BatchItem[](3);
+        items[0] = IEVC.BatchItem({
+            onBehalfOfAccount: address(0),
+            targetContract: address(evc),
+            value: 0,
+            data: abi.encodeWithSelector(evc.enableController.selector, user1, address(eTST2))
+        });
+        items[1] = IEVC.BatchItem({
+            onBehalfOfAccount: user1,
+            targetContract: address(eTST2),
+            value: 0,
+            data: abi.encodeWithSelector(eTST2.borrow.selector, 10e18, user1)
+        });
+        items[2] = IEVC.BatchItem({
+            onBehalfOfAccount: user1,
+            targetContract: address(eTST2),
+            value: 0,
+            data: abi.encodeWithSelector(eTST2.accountLiquidity.selector, user1, false)
+        });
+
+        startHoax(user1);
+        (IEVC.BatchItemResult[] memory batchItemsResult, IEVC.StatusCheckResult[] memory accountsStatusCheckResult,) =
+            evc.batchSimulation(items);
+
+        assertEq(accountsStatusCheckResult[0].result, abi.encodeWithSignature("E_AccountLiquidity()"));
+        (uint256 collateralValue, uint256 liabilityValue) = abi.decode(batchItemsResult[2].result, (uint256, uint256));
+
+        // health score < 1
+        assertEq(collateralValue * 1e18 / liabilityValue, 0.75e18);
+    }
+
+    //batch simulation executes all items
+    function test_batchSimulation_executesAllItems() public {
+        IEVC.BatchItem[] memory items = new IEVC.BatchItem[](3);
+        items[0] = IEVC.BatchItem({
+            onBehalfOfAccount: user1,
+            targetContract: address(eTST2),
+            value: 0,
+            data: abi.encodeWithSelector(eTST2.borrow.selector, 0.1e18, user1)
+        });
+        items[1] = IEVC.BatchItem({
+            onBehalfOfAccount: user1,
+            targetContract: address(assetTST),
+            value: 0,
+            data: abi.encodeWithSignature(
+                "permit(address,address,uint256,uint256,uint8,bytes32,bytes32)",
+                address(0),
+                address(0),
+                type(uint256).max,
+                0,
+                0,
+                bytes32("0"),
+                bytes32("0")
+            )
+        });
+        items[2] = IEVC.BatchItem({
+            onBehalfOfAccount: user1,
+            targetContract: address(eTST2),
+            value: 0,
+            data: abi.encodeWithSelector(eTST2.borrow.selector, 10e18, user1)
+        });
+
+        startHoax(user1);
+        (IEVC.BatchItemResult[] memory batchItemsResult,,) = evc.batchSimulation(items);
+
+        assertEq(batchItemsResult[1].success, false);
+        bytes memory err = abi.encodeWithSignature("Error(string)", "permit: invalid signature");
+        assertEq(batchItemsResult[1].result, err);
+    }
+
+    function getSubAccount(address primary, uint8 subAccountId) internal pure returns (address) {
```

Comment:
copies can be removed if the helper is moved to the top contract (EVaultTestBase)